File: app/src/main/java/com/aistra/hail/utils/AppMetadataEntity.java
package com.aistra.hail.utils;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "app_metadata")
public class AppMetadataEntity {
    @PrimaryKey
    @NonNull
    public String packageName = "";
    public String name = "";
    public boolean systemApp;
    public long firstInstallTime;
    public long lastUpdateTime;
    public int flags;
    public boolean enabled;
    public boolean installed;
    @NonNull
    public String sourceSignature = "";
}

File: app/src/main/kotlin/com/aistra/hail/utils/AppMetadataEntity.kt
package com.aistra.hail.utils

import androidx.room3.ColumnInfo
import androidx.room3.Entity
import androidx.room3.PrimaryKey

@Entity(tableName = "app_metadata")
data class AppMetadataEntity(
    @PrimaryKey val packageName: String = "",
    val name: String = "",
    val systemApp: Boolean = false,
    val firstInstallTime: Long = 0L,
    val lastUpdateTime: Long = 0L,
    val flags: Int = 0,
    val enabled: Boolean = false,
    val installed: Boolean = false,
    @ColumnInfo(defaultValue = "0") val frozen: Boolean = false,
    val sourceSignature: String = ""
)
